### PR TITLE
Extract and document Boxel::IconButton

### PR DESCRIPTION
**Pull request by lukemelia**
Screenshot 2021-02-03 at 1 13 13 AM

Screenshot 2021-02-03 at 1 14 13 AM

### CODE REVIEW

File: app/css/components/next-steps.css [hunk 1]
```diff
@@ -1,8 +1,8 @@
 .next-steps {
   padding: 30px;
   background-color: #41404D;
-  color: var(--boxel-light);
-  font: var(--boxel-font-sm);
+  color: var(--boxel-utloight);
+  font: var(--boxel-utfoont-sm);
```
Comment (burieberry) [inline app/css/components/next-steps.css]: Seeing some find + replace victims here

**Comment by burieberry** [comment]
Also noticed that the base styling is overriding the use case styling here. Might have to do something like `.discrepancy-field__btn-container > .discrepancy-field__btn` in the discrepancy css file.

css-order

File: app/css/components/next-steps.css (hunk 1 above)
Comment (lukemelia) [reply app/css/components/next-steps.css]: Fixed, or should i say "utfoixed"

**Comment by lukemelia** [comment]
> Also noticed that the base styling is overriding the use case styling here. Might have to do something like .discrepancy-field__btn-container > .discrepancy-field__btn in the discrepancy css file.

After considering a few options, I introduced a new component to avoid this.

File: app/components/boxel/icon-button/usage.hbs [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+  </:example>
+
+  <:api as |Args|>
+    <Args.String @name="icon"
+      @description='the name of the svg to show'
+      @value={{this.icon}}
+      @onInput={{fn (mut this.icon)}}
+    />
+    <Args.String @name="variant"
+      @description='the variant to render as (applies CSS class) - `null` or `"primary"` or `"secondary"`';
```
Comment (mattmcmanus) [inline app/components/boxel/icon-button/usage.hbs:22]: Seems like passing in @options would be preferable here?

Comment (lukemelia) [reply app/components/boxel/icon-button/usage.hbs:22]: @mattmcmanus I wasn't sure how we would deal with null in that case, which is the most common and default value

**Review by burieberry — APPROVED** [review]

**Pull request merged**